src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers/ClientMethodsAnalyzer.cs

```diff
@@ -21,104 +28,126 @@ public class ClientMethodsAnalyzer : ClientAnalyzerBase
             Descriptors.AZC0015
         });
 
-        private static void CheckClientMethod(ISymbolAnalysisContext context, IMethodSymbol member)
+        private static void CheckCancellationTokenOrRequestContext(ISymbolAnalysisContext context, IMethodSymbol member)
         {
-            static bool SupportsCancellationsParameter(IParameterSymbol parameterSymbol)
+            static bool IsRequestContext(IParameterSymbol parameterSymbol)
```

Comment:
Do these need to be local functions?  I think it would make the code more readable (and reusable) if they were declared on the type.